core/org.eclipse.cdt.core/model/org/eclipse/cdt/internal/core/settings/model/CProjectDescriptionDelta.java

```diff
@@ -74,7 +74,7 @@ public ICSettingObject getSetting() {
 
 	@Override
 	public int getSettingType() {
-		return fSetting.getType();
+		return fSetting == null ? 0 : fSetting.getType();
```

Comment:
What does `0` mean here?
Is it safe to always return this value in the absence of `fSetting`?

Reply:
The int is a flag where each bit is an attribute that is (not)set.
0 as such means no flags set.
Is this safe? I think so because fSetting is never null in the current code.
If fSettings is null the current code throws an exception.